src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -91,21 +81,94 @@ const DiffExprCompute = (props) => {
       comparisonGroup[selectedComparison]['basis'] = `sample/${samples[0].key}`
     }
 
+    setSampleKeys(samples.map(sample => sample.key));
+
   }, [hierarchy, properties]);
 
-  const validateForm = () => {
+  const cellIdToSampleMap = useMemo(() => {
+    const mapping = [];
+    sampleKeys.forEach((key, idx) => {
+      const cellIds = properties[key].cellIds;
+      cellIds.forEach(cellId => mapping[cellId] = idx);
+    });
 
-    if (!comparisonGroup[selectedComparison]?.cellSet) {
-      setIsFormValid(false);
-      return;
+    return mapping;
+  }, [numSamples]);
+
+  // This function will return true if each of the compared groups is made up of at least
+  // 1 sample of with a enough cells (> MIN_NUM_CELLS_IN_GROUP).
+  const canRunDiffExpr = useCallback(() => {
+
+    if (selectedComparison === ComparisonType.WITHIN) return true;
+
+    if (!basis || !cellSet || !compareWith || !cellIdToSampleMap.length > 0) { return false; }
+
+    const basisCellSetKey = getCellSetKey(basis);
+
+    const cellSetKey = getCellSetKey(cellSet);
+    const compareWithKey = getCellSetKey(compareWith);
+
+    let basisCellSet = [];
+    if (basisCellSetKey === 'all') {
+      const allCellIds = sampleKeys.reduce((cumulativeCellIds, key) => {
+        const cellIds = properties[key].cellIds;
+        return cumulativeCellIds.concat(Array.from(cellIds));
+      }, []);
+      basisCellSet = new Set(allCellIds);
+    } else {
+      basisCellSet = properties[basisCellSetKey].cellIds;
+    }
+
+    const cellSetCellIds = Array.from(properties[cellSetKey].cellIds);
+
+    let compareWithCellIds = [];
+    if (['rest', 'background'].includes(compareWithKey)) {
+      const parentGroup = getRootKey(cellSet);
+
+      const otherGroupKeys = hierarchy.find(obj => obj.key === parentGroup)
+        .children.filter(child => child.key !== cellSetKey);
+
+      compareWithCellIds = otherGroupKeys.reduce((acc, child) => {
```

Comment:
to be consistent with the rest of the naming, this should be `compareWithCellSetIds`

Reply:
This is the `ids` of the `cell`, so it should be `...cellIds`, not the `ids` of the cell set.